src/lib.rs

```diff
@@ -158,12 +158,15 @@ cfg_if! {
         pub type uintptr_t = usize;
         pub type ssize_t = isize;
 
+        #[cfg(not(target_env = "sgx"))]
         pub enum FILE {}
+        #[cfg(not(target_env = "sgx"))]
         pub enum fpos_t {} // TODO: fill this out with a struct
 
         pub const INT_MIN: c_int = -2147483648;
         pub const INT_MAX: c_int = 2147483647;
 
+        #[cfg(not(target_env = "sgx"))]
```

Comment:
Same for these (use `cfg_if`). 

Does this mean that `sgx` does not have `strlen`, `memcpy`, etc ?

Reply:
SGX doesn't have a `libc`. The definitions in here are basically just what would be in a hypothetical `ctypes` crate.

`memcpy` is provided by `compiler_builtins`

I have at times compiled a static library with some of the C functions (strcpy, etc.) to link to existing C libraries and an implementation of `malloc` in terms of the Rust allocator. Would it makes sense to add these to this crate (behind a feature)?

Reply:
In order to add a target to `std`, it is currently required to make changes to `libc`.

> `libstd` depends on the target having a full-featured `libc`.

No it doesn't, `libstd` depends on exactly those features it uses from `libc`. In case of the SGX port, that is none.

> Also, this crate links against a `libc` library.

No it doesn't, it expects the linker to insert this dependency automatically. This doesn't happen in the SGX port.

All this basically works the same as in wasm32-unknown-unknown.

Reply:
> All this basically works the same as in wasm32-unknown-unknown.

`libc` should not support `wasm32-unknown-unknown` because it does not have a `libc` implementation.